stable_baselines3/common/utils.py

```diff
@@ -92,7 +92,7 @@ def get_schedule_fn(value_schedule: Union[Schedule, float]) -> Schedule:
         value_schedule = constant_fn(float(value_schedule))
     else:
         assert callable(value_schedule)
-    return value_schedule
+    return lambda _: float(value_schedule(_))
```

Comment:
this won't work, maybe a better solution is to do a call to `value_schedule(1.0)` and check that the return type is a float (and output a useful error message if not).